packages/functional-tests/tests/react-conversion/oauthResetPassword.spec.ts

```diff
@@ -0,0 +1,315 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+import { test, expect } from '../../lib/fixtures/standard';
+import { EmailHeader, EmailType } from '../../lib/email';
+
+test.describe.only('oauth reset password react', () => {
+  let resetPasswordReactFlag = false;
+
+  test.beforeEach(async ({ pages: { login, resetPassword } }) => {
+    resetPasswordReactFlag = resetPassword.react;
+    resetPassword.react = true;
+    test.slow();
+
+    const config = await login.getConfig();
+    test.skip(config.showReactApp.resetPasswordRoutes !== true);
+    test.skip(config.showReactApp.oauthRoutes !== true);
+  });
+
+  test.afterEach(async ({ pages: { resetPassword } }) => {
+    resetPassword.react = resetPasswordReactFlag;
+  });
+
+  test('reset password', async ({
+    target,
+    page,
+    credentials,
+    pages: { login, relier, resetPassword, settings },
+  }) => {
+    await openRelier({ pages: { settings, relier } });
+    await passwordResetFlow(
+      {
+        target,
+        page,
+        credentials,
+        pages: { login, relier, resetPassword },
+      },
+      {}
+    );
+  });
+
+  test('reset password different tab', async ({
+    target,
+    page,
+    credentials,
+    pages: { login, relier, resetPassword, settings },
+  }) => {
+    await openRelier({ pages: { relier, settings } });
+    await passwordResetFlow(
+      {
+        target,
+        page,
+        credentials,
+        pages: { login, relier, resetPassword },
+      },
+      {
+        newTab: true,
+      }
+    );
+  });
+
+  test('reset password scoped keys', async ({
+    target,
+    page,
+    credentials,
+    pages: { login, relier, resetPassword, settings },
+  }) => {
+    await openRelier({ pages: { relier, settings } });
+    await passwordResetFlow(
+      {
+        target,
+        page,
+        credentials,
+        pages: { login, relier, resetPassword },
+      },
+      {
+        action: 'scopedKeys',
+      }
+    );
+  });
+
+  test('reset password with PKCE different tab', async ({
+    target,
+    page,
+    credentials,
+    pages: { login, relier, resetPassword },
+  }) => {
+    await passwordResetFlow(
+      {
+        target,
+        page,
+        credentials,
+        pages: { login, relier, resetPassword },
+      },
+      {
+        action: 'pkce',
+        newTab: true,
```

Comment:
Note, that until we fix that weird cors error, we can't test the same tab flow. Essentially the cors error stops us from getting the rp's 'state' which is needed to complete the full flow on same tab.